freqtrade/plugins/pairlist/RemotePairList.py

```diff
@@ -0,0 +1,166 @@
+from freqtrade.plugins.pairlist.IPairList import IPairList
+
+
+logger = logging.getLogger(__name__)
+
+
+class RemotePairList(IPairList):
+
+    def __init__(self, exchange, pairlistmanager,
+                 config: Config, pairlistconfig: Dict[str, Any],
+                 pairlist_pos: int) -> None:
+        super().__init__(exchange, pairlistmanager, config, pairlistconfig, pairlist_pos)
+
+        if 'number_assets' not in self._pairlistconfig:
+            raise OperationalException(
+                '`number_assets` not specified. Please check your configuration '
+                'for "pairlist.config.number_assets"')
+
+        if 'pairlist_url' not in self._pairlistconfig:
+            raise OperationalException(
+                '`pairlist_url` not specified. Please check your configuration '
+                'for "pairlist.config.pairlist_url"')
+
+        self._number_pairs = self._pairlistconfig['number_assets']
+        self._refresh_period = self._pairlistconfig.get('refresh_period', 1800)
+        self._keep_pairlist_on_failure = self._pairlistconfig.get('keep_pairlist_on_failure', True)
+        self._pair_cache: TTLCache = TTLCache(maxsize=1, ttl=self._refresh_period)
+        self._pairlist_url = self._pairlistconfig.get('pairlist_url', '')
+        self._read_timeout = self._pairlistconfig.get('read_timeout', 60)
+        self._bearer_token = self._pairlistconfig.get('bearer_token', '')
+        self._last_pairlist: List[Any] = list()
+
+    @property
+    def needstickers(self) -> bool:
+        """
+        Boolean property defining if tickers are necessary.
+        If no Pairlist requires tickers, an empty Dict is passed
+        as tickers argument to filter_pairlist
+        """
+        return False
+
+    def short_desc(self) -> str:
+        """
+        Short whitelist method description - used for startup-messages
+        """
+        return f"{self.name} - {self._pairlistconfig['number_assets']} pairs from RemotePairlist."
+
+    def fetch_pairlist(self) -> Tuple[List[str], float, str]:
+
+        headers = {
+            'User-Agent': 'Freqtrade/' + __version__ + ' Remotepairlist'
+        }
+
+        if self._bearer_token:
+            headers['Authorization'] = f'Bearer {self._bearer_token}'
+
+        info = "Pairlist"
+
+        try:
+            response = requests.get(self._pairlist_url, headers=headers,
+                                    timeout=self._read_timeout)
+            content_type = response.headers.get('content-type')
+            time_elapsed = response.elapsed.total_seconds()
+
+            if "application/json" in str(content_type):
+                jsonparse = response.json()
+                pairlist = jsonparse['pairs']
+                info = jsonparse.get('info', '')
+            else:
+                raise OperationalException('RemotePairList is not of type JSON abort ')
+
+            self._refresh_period = jsonparse.get('refresh_period', self._refresh_period)
```

Comment:
i'm not sure i like this - do we really want a remote source (a source we don't control) to update our cache settings?
it can be the maximum frequency allowed - but just because a source allows a refresh every minute doesn't mean that's what i want for my strategy.

Reply:
The idea was that a remotesource maybe would not like it to have the pairlist refreshed every 1 minute, if the remote pairlist for example only updates every 30 minutes, since that would be a waste of resources and no point, thinking about it the better way would be to allow the minimum refresh period to be set from the remote end and if its higher keep the local setting.

Reply:
i'm not against changing this per se (if the config doesn't specify a refresh_period, for example) - but i think the "final say" should be with the strategy - at least if it's higher there.

just because an api allows a request every minute doesn't mean i actually want a new pairlist every minute - so i think the "this defines the minimum - but use the config setup if it's higher" will be best.

Reply:
the refresh_period is now increased to the one sent from the server (minimum) if it changes and logged